lib/util/include/hse_util/compiler.h

```diff
@@ -110,12 +104,18 @@
 #endif
 
 #ifdef SUPPORTS_ATTR_NONNULL
-/* HSE_NONNULL(...) cannot go at the end of the function declaration */
+/* HSE_NONNULL(...) cannot go at the end of the function definition */
 #define HSE_NONNULL(...)        __attribute__((__nonnull__(__VA_ARGS__)))
 #else
 #define HSE_NONNULL(...)
 #endif
 
+#if HSE_MOCKING
+#define hse_static              HSE_WEAK
```

Comment:
Using this in order to override the params_size() function defined in param.c

Reply:
That's truly confusing, does "hse_weak" mean it's static? 😄 

Seems like hse_static should just go away, or be something like MTF_STATIC which is a common idiom.  We have many functions that are exposed only so that unit tests can call them, and that hurts the ability of the compiler to optimize the code.  All those functions should be marked "MTF_STATIC" (or whatever).

Reply:
All I am saying is I need access to that function to override it.

```c
#if HSE_MOCKING
HSE_WEAK
#else
static
#endif
param_size(...);
```

Something tells me you won't approve the above.

Reply:
Well, it is rather gruesome 😄 Is this just a one-off, or a pattern to be applied elsewhere?  Are you using weak to avoid the MTF_MOCK() macros?

Reply:
I don't think there is precedence for MTF_MOCK in C files. This pattern would be useful elsewhere.

Reply:
Sorry, I meant the MOCK_SET()/MOCK_UNSET() macros that will redirect a function call to a mock.  Can you use that instead of weak?  I'm not clear on the application...

Reply:
I would use that, but I don't want to expose this function outside of the compilation unit.

Reply:
Is this addressing the issue of being forced to make symbols non-static in order to access them from unit tests?

Reply:
We should have done this long ago, either HSE_STATIC or MTF_STATIC (I like the latter as it implies the who and why it's conditionally static).